android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt

```diff
@@ -411,11 +412,21 @@ class NetworkIntentService : IntentService("NetworkIntentService") {
             Log.d(TAG, "create request to " + captive_portal_url + " for url: " + envoyUrl)
         }
 
+        // set up http cache dir
+        val cacheDir = File(applicationContext.cacheDir, envoyService + "-cache")
+        if (!cacheDir.exists()) {
+            Log.d(TAG, "cache setup, create " + cacheDir.absolutePath)
+            cacheDir.mkdirs()
+        }
+
         val executor: Executor = Executors.newSingleThreadExecutor()
         val myBuilder = CronetEngine.Builder(applicationContext)
         val cronetEngine: CronetEngine = myBuilder
             .setEnvoyUrl(envoyUrl)
-            .setUserAgent(DEFAULT_USER_AGENT).build()
+            .setStoragePath(cacheDir.absolutePath)
+            .enableHttpCache(CronetEngine.Builder.HTTP_CACHE_DISK, 1 * 1024 * 1024) // 1 megaByte
```

Comment:
how did you pick 1MB?

Reply:
i just reduced it from the 10mb cache used elsewhere since all it's expected to cache is the google 204 response

Reply:
and presumably it's also used for DNS responses, but 1MB seems fine for the test